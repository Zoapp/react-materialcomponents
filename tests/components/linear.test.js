import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import LinearProgress from "libs/components/linearProgress";

describe("components/LinearProgress", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<LinearProgress />);
  });

  it("can be indeterminate", () => {
    wrapper.setProps({ indeterminate: true });
    expect(wrapper.hasClass("mdc-linear-progress--indeterminate")).toEqual(
      true,
    );
  });

  it("can be closed", () => {
    wrapper.setProps({ closed: true });
    expect(wrapper.hasClass("mdc-linear-progress--closed")).toEqual(true);
  });

  it("can be reversed", () => {
    wrapper.setProps({ reversed: true });
    expect(wrapper.hasClass("mdc-linear-progress--reversed")).toEqual(true);
  });

  it("can render a presetted progress", () => {
    const component = renderer.create(<LinearProgress progress={50} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a buffer", () => {
    const component = renderer.create(<LinearProgress buffer={50} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
