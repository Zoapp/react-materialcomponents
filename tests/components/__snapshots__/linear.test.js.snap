// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LinearProgress can have a buffer 1`] = `
<div
  className="mdc-linear-progress"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "transform": "scaleX(1)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "transform": "scaleX(0)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`components/LinearProgress can render a presetted progress 1`] = `
<div
  className="mdc-linear-progress"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "transform": "scaleX(0)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "transform": "scaleX(1)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;
